lib/centralisation/providers/favorites_providers.dart

```diff
@@ -27,6 +25,17 @@ class FavoritesNameNotifier extends StateNotifier<List<String>> {
     saveFavoritesToSharedPreferences(state);
   }
 
+  void reorderFavorites(int oldIndex, int newIndex) {
+    if (oldIndex < newIndex) {
+      newIndex -= 1;
+    }
+    final copy = state.sublist(0);
```

Comment:
There is no use of passing by copy, every action can be done on state if you end with state = state

Reply:
This does not set the state of the provider properly, it does trigger the provider to be rebuilt since it is modified more than once (with .insert and =)